securedrop/journalist.py

```diff
@@ -408,20 +580,16 @@ def remove_star(sid):
     return redirect(url_for('index'))
 
 
-@app.route('/')
-@login_required
-def index():
+def get_stars_and_labels(sources):
+    """Get the stars and labels for sources"""
     unstarred = []
     starred = []
+    source_labels = {}
+    all_source_labels = get_source_tags_in_use()
 
-    # Long SQLAlchemy statements look best when formatted according to
-    # the Pocoo style guide, IMHO:
-    # http://www.pocoo.org/internal/styleguide/
-    sources = Source.query.filter_by(pending=False) \
-                          .order_by(Source.last_updated.desc()) \
-                          .all()
     for source in sources:
         star = SourceStar.query.filter_by(source_id=source.id).first()
+        source_labels.update({source.id: get_source_tags(source.id)})
```

Comment:
Yep, totally. I've removed the query I added to this for loop and instead am using the `tags` attribute I added to Source and Submission models (using SQLAlchemy's `relationship` directive) to get the tags for the source.